### ISSUE TITLE
Pyrebase cant auth vs realtime DB rules

### ISSUE THREAD

**Issue by DizzyduckAR** [thisbejim/Pyrebase#341]
unless DB rules are
".read": true,
".write":true
it will not let you pass auth the users UID to ruler.

while the auth signup and the UID grab working correct any try to pass it to the DB server ruler fails.
it will not let it read or write.
testing same rules with simulator grant correct access. running it live on client fail to write. 
(write the email and pass in auth. grant uid. fail to write into db)

example fail to pass:
{
  "rules": {
      ".read": "auth != null",
      ".write": "auth != null"
}}

so atm we got no way to auth users on real time DB
![chrome_kfffjXQJXw](https://user-images.githubusercontent.com/52171360/80585706-a56bee80-89c8-11ea-915e-22237654b772.png)

**Comment by DafzNet** [comment]
Same thing Im facing now. Found a solution?

**Comment by DizzyduckAR** [comment]
Info:
DB can be Fully Locked.
on firebase cli you got "cloud function"
this tab allow you to write and push C funcs to your project.
once hooked you can run deploy and it will give you a link.

C funcs bypass any DB rules and can run as Admin.(while the func on the server itself)
Demo On create event.
![Code_N54Hs8teYS](https://user-images.githubusercontent.com/52171360/92994581-6ab6ae80-f4b0-11ea-85b6-37800ad17c8c.png)

so the code above is uploaded to the main project.
when user created in auto it will pass the uid to new DB and write some values.
it will do it as admin on the server itself.
you can easy expend this by using Http Req func that will allow you to send and receive data fully protected while no auth runs local.

bottom line no matter what the bug on the DB rules you can bypass it with cloud funcs and it will be much much much safer.

yup.
you need to cloud func the auth and db write into cloud func in firebase.
it will bypass the issue this bug made.
![Code_OSjW2p42Lg](https://user-images.githubusercontent.com/52171360/92446820-3fa82400-f16b-11ea-9dfc-158a6f9e3c43.png)
![Code_fmsHh7JskM](https://user-images.githubusercontent.com/52171360/92446831-43d44180-f16b-11ea-9cf8-f526034e7d8c.png)


also made a blank ready ver with qt and pyrebase if you need

**Cfunc can be written in js and typescript (it got admin priv since its run on the server itself

**Comment by dantebarba** [comment]
Any solution to this? Auth is not working for me neither.
